packages/adobe-sign/__tests__/apis/adobe-sign-api.spec.ts

```diff
@@ -0,0 +1,75 @@
+import "reflect-metadata";
+import { AdobeSignApi } from '../../src/apis/adobe-sign-api';
+import {AxiosProvider, SecretsManagerService} from "asu-core";
+import {Mock} from "asu-core/src/testing/mock-provider";
+import {Axios} from "axios";
+import {InternalApiError} from "../../src/apis/errors/internal-api-error";
+import {AdobeApiError} from "../../src/apis/errors/adobe-api-error";
+
+describe('AdobeSignAPI', () => {
+    function setup() {
+        const secretsManagerService = Mock(new SecretsManagerService(null));
+        const mockAxios = Mock(new Axios());
+        const axiosProvider = Mock(new AxiosProvider())
+        axiosProvider.resolve.mockReturnValue(mockAxios);
+        const adobeApi = new AdobeSignApi(secretsManagerService, axiosProvider);
+
+        return { secretsManagerService, adobeApi, mockAxios };
+    }
+
+    describe('getAgreement', () => {
+        it('should be called with the given id and return Agreement object', async () => {
+            // Arrange
+            let { adobeApi, mockAxios } = setup();
+            const mockData = {
+                id: 'CBJCHBCAABAAN24SWUnGHW-o_NaT5i3O5lKuHiccQ2GP',
+                name: '[DEMO USE ONLY] Agreement Test 001',
+                groupId: 'CBJCHBCAABAACnRhBD9pKDAfZ55583n-4WOlwlvI_RGM',
+                type: 'AGREEMENT',
+                status: 'SIGNED',
+                hasFormFieldData: true,
+            }
+
+            mockAxios.get.mockResolvedValue({data: mockData});
+            let apiSpy = jest.spyOn(adobeApi, 'getAgreement')
```

Comment:
It looks like you are spying on the function that you are calling a few lines down. You are essentially asserting that your test is calling `adobeApi.getAgreement`.